### ISSUE TITLE
Incorrect warning for arguments are thrown on static arrow functions

### ISSUE THREAD

**Issue by arkener** [sirbrillig/phpcs-variable-analysis#275]
Related to #272, incorrect warnings for arguments are thrown on static arrow functions as stated on https://www.drupal.org/project/coder/issues/3303720#comment-14652996

```php
<?php

function lorem() {
  $foo = static fn (string $value) => $value;
  echo $foo('bar');
}
```

Will result in the following:
```
FILE: test.php
-----------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------------------------------------
 4 | WARNING | Unused static variable $value. (VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable)
 4 | WARNING | Redeclaration of function parameter $value as static variable.
   |         | (VariableAnalysis.CodeAnalysis.VariableAnalysis.VariableRedeclaration)
-----------------------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by sirbrillig** [comment]
🤦 Ah, I didn't think of static arrow functions. I don't think I've seen one before!
